src/solve.jl

```diff
@@ -75,7 +75,7 @@ function init(prob::AbstractDDEProblem{uType,tType,lType,isinplace}, alg::algTyp
                                                        ode_prob,
                                                        OrdinaryDiffEq.alg_order(alg)))
     end
-    integrator.dt = dt
+    integrator.dt = zero(dt)
```

Comment:
> But almost all of the interpolations have u0 + dt* ..., so it'll "accidentally" work out as a constant extrapolation.

I guess it won't, and it didn't in my tests with BS3 and Tsit5. The problem is that the expression in brackets after `u0 + dt*` is not a number in these cases; an example is line https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/dense/interpolants.jl#L123.